src/Workspaces/Core/MSBuild/MSBuild/MSBuildProjectLoader.Worker.cs

```diff
@@ -408,10 +408,16 @@ private IEnumerable<AnalyzerReference> ResolveAnalyzerReferences(CommandLineArgu
 
                 foreach (var path in commandLineArgs.AnalyzerReferences.Select(r => r.FilePath))
                 {
-                    var fullPath = Path.GetFullPath(path);
-                    if (File.Exists(fullPath))
+                    string fullPath;
+
+                    if (PathUtilities.IsAbsolute(path))
                     {
-                        analyzerLoader.AddDependencyLocation(fullPath);
+                        fullPath = FileUtilities.TryNormalizeAbsolutePath(path);
+
+                        if (fullPath != null && File.Exists(fullPath))
```

Comment:
Does File.Exists throw exceptions? The documentation doesn't list and and the [source](https://referencesource.microsoft.com/#mscorlib/system/io/file.cs,416) appears to catch exceptions.

Reply:
Hmm - I thought it did, but it's possible that everything I believe is a lie.

Reply:
Even so, I expect that `File.Exists` would throw the same exceptions for the same circumstances as `Path.GetFullPath(...)`. Those are covered by the call to `FileUtilities.TryNormalizeAbsolutePath(...)` above because that API simply calls `Path.GetFullPath(...)` and returns null if any exceptions are thrown.

Reply:
Yeah, I think it's probably safe. These all print False to the Console

```C#
Console.WriteLine(File.Exists(null));
Console.WriteLine(File.Exists("COM1"));
Console.WriteLine(File.Exists(".."));
``